templates/react-with-storybook/.storybook/main.js

```diff
@@ -1,4 +1,7 @@
 module.exports = {
   stories: ['../stories/**/*.stories.@(ts|tsx|js|jsx)'],
   addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
+  typescript: {
```

Comment:
```suggestion
  // https://storybook.js.org/docs/react/configure/typescript#mainjs-configuration
  typescript: {
```
Add a link to the docs for this config as we did with `preview.js` in #805 ; think docs links like these are generally useful, particularly for examples like the templates.